app/questionnaire/questionnaire_schema.py

```diff
@@ -1375,16 +1313,16 @@ def _get_rules_section_dependencies(
                 dependencies_for_progress_value_source["blocks"]
             )
 
-            # Type Ignore: Added to this method as the block will exist at this point
-            for answer_id in answer_id_list:
-                block = self.get_block_for_answer_id(answer_id)  # type: ignore
-                section_id = self.get_section_id_for_block_id(block["id"])  # type: ignore
+            section_dependencies |= (
+                self._get_section_dependencies_for_dependent_answers(
+                    current_section_id, dependent_answer_ids
+                )
+            )
 
-                if section_id != current_section_id:
-                    self._when_rules_section_dependencies_by_answer[answer_id].add(
-                        current_section_id
-                    )
-                    section_dependencies.add(section_id)  # type: ignore
+            for list_name in dependent_list_names:
```

Comment:
The previous one was ignoring list dependencies for rules in `["question_variants", "content_variants"]` keys, now I'm wondering why.

Reply:
Going to try an example as I'm also not sure why those would have been skipped

Reply:
It looks like this was probably done for performance benefit - the question variants have the same question-ids, block-ids, and answer-ids so the progress is not going to be affected by adding or removing to a list, the changed answer will just relate to a question/answer with different text etc. 
In which case I could probably do the same thing? It saves a few checks but are there edge cases where we do want to check those sections? currently i cant think of any so ignoring makes sense

Reply:
Since it is not done at run time anymore, you can't ignore it without making it work at runtime or having two maps?. As only it shouldn't be ignored in all places?

Reply:
Actually, it shouldn't matter, it doesn't affect completeness therefore should be safe to ignore variants. We don't reshow pages if "placeholders" change, this would be the same concept. Until they become a requirement, we can ignore variants.

Reply:
Have ignored them for when rules and added an extra test for it